services/common/rs_server_common/utils/utils.py

```diff
@@ -312,3 +337,54 @@ def eodag_download(argument: EoDAGDownloadHandler, db, init_provider: Callable[[
     # Try n times to update the status to DONE in the database
     update_db(db, db_product, EDownloadStatus.DONE)
     logger.debug("Download finished succesfully for %s", db_product.name)
+
+
+def odata_to_stac(feature_template: dict, odata_dict: dict, odata_stac_mapper: dict) -> dict:
+    """This function is used to map odata values to a given STAC template"""
+    if not all(item in feature_template.keys() for item in ["properties", "id", "assets"]):
+        raise ValueError("Invalid stac feature template")
+    for stac_key, eodag_key in odata_stac_mapper.items():
+        if eodag_key in odata_dict:
+            if stac_key in feature_template["properties"]:
+                feature_template["properties"][stac_key] = odata_dict[eodag_key]
+            elif stac_key == "id":
+                feature_template["id"] = odata_dict[eodag_key]
+            elif stac_key == "file:size":
+                feature_template["assets"]["file"][stac_key] = odata_dict[eodag_key]
+    return feature_template
+
+
+def extract_eo_product(eo_product: EOProduct, mapper: dict) -> dict:
+    """This function is creating key:value pairs from an EOProduct properties"""
+    return {key: value for key, value in eo_product.properties.items() if key in mapper.values()}
+
+
+def create_stac_collection(products: List[EOProduct], feature_template: dict, stac_mapper: dict) -> dict:
+    """This function create a STAC feature for each EOProduct based on a given template"""
+    stac_template: Dict[Any, Any] = {
+        "type": "FeatureCollection",
+        "numberMatched": 0,
+        "numberReturned": 0,
+        "features": [],
+    }
+    for product in products:
+        product_data = extract_eo_product(product, stac_mapper)
+        feature_tmp = odata_to_stac(copy.deepcopy(feature_template), product_data, stac_mapper)
+        stac_template["numberMatched"] += 1
+        stac_template["numberReturned"] += 1
+        stac_template["features"].append(feature_tmp)
+    return stac_template
+
+
+def sort_feature_collection(feature_collection: dict, sortby: str = "+datetime") -> dict:
+    """This function sorts a STAC feature collection based on a given criteria"""
+    # Force default sorting even if the input is invalid, don't block the return collection because of sorting.
+    order = "+" if sortby[0] not in ["+", "-"] else sortby[0]
+    if len(feature_collection["features"]) and hasattr(feature_collection["features"][0], "properties"):
```

Comment:
Instead of hasattr(feature_collection["features"][0], "properties")
I think it should be "properties" in feature_collection["features"][0]